### DISCUSSION TITLE
Custom icons from web info?

### DISCUSSION THREAD

**Discussion by Atrello** [filips123/PWAsForFirefox#139] [Q&A] [answered]
Is there a way to use a websites .ico favicon as its icon?

**Comment by filips123** [comment, marked as answer]
PWAsForFirefox will use icons from a [web app manifest](https://developer.mozilla.org/en-US/docs/Web/Manifest/icons) provided by a website, or `link[rel=icon]` links if the manifest is not provided.

Is there any reason why you specifically want `.ico` favicons? They are not very common these days, as most websites specify icons in at least one of two ways I mentioned above.

---

If you want to set a custom PWA icon, this varies depending on your OS:

* On Windows, icons are stored in `%APPDATA%\FirefoxPWA\icons\SITE_ID` (these are actual `.ico` files because Windows requires them). 
* On Linux, they are stored in subdirectories of `~/.local/share/icons/hicolor` (or `$XDG_DATA_HOME/icons/hicolor`, if `$XDG_DATA_HOME` is set), for example, `~/.local/share/icons/hicolor/ICON_SIZE/apps/FFPWA-SITE_ID.png` . Note that one PWA may have multiple icons with different sizes, so you need to replace (or remove) all of them to make sure your custom icon shows correctly.
* On macOS, icons are stored inside app bundle, as a ICNS file: `~/Applications/SITE_NAME.app/Contents/Resources/app.icns`.

To get the site ID, run `firefoxpwa profile list`.

I will probably implement support for setting icons using extension sometime in the future (see #128), but unfortunately I currently don't have time.

**Comment by Atrello** [comment]
Thanks very much
